providers/cloudns/cloudnsProvider.go

```diff
@@ -388,6 +409,27 @@ func toReq(rc *models.RecordConfig) (requestParams, error) {
 		req["algorithm"] = strconv.Itoa(int(rc.DsAlgorithm))
 		req["digest-type"] = strconv.Itoa(int(rc.DsDigestType))
 		req["record"] = rc.DsDigest
+	case "NAPTR":
+		req["order"] = strconv.Itoa(int(rc.NaptrOrder))
+		req["pref"] = strconv.Itoa(int(rc.NaptrPreference))
+		req["flag"] = rc.NaptrFlags
+		req["params"] = rc.NaptrService
+		req["regexp"] = rc.NaptrRegexp
+		req["replace"] = rc.GetTargetField()
+	case "LOC":
+		parts := strings.Fields(rc.GetTargetCombined())
+		req["lat-deg"] = parts[0]
+		req["lat-min"] = parts[1]
+		req["lat-sec"] = parts[2]
+		req["lat-dir"] = parts[3]
+		req["long-deg"] = parts[4]
+		req["long-min"] = parts[5]
+		req["long-sec"] = parts[6]
+		req["long-dir"] = parts[7]
+		req["altitude"] = strings.Split(parts[8], "m")[0]
```

Comment:
And this.

Reply:
I'll wait for your change to be merged then rebase this. Thanks.